gobblin-runtime/src/main/java/gobblin/runtime/TaskState.java

```diff
@@ -169,6 +169,16 @@ public void setTaskDuration(long duration) {
   }
 
   /**
+   * Return whether the task has completed running or not.
+   *
+   * @return {@code true} if the task has completed or {@code false} otherwise
+   */
+  public boolean isCompleted() {
+    WorkingState state = getWorkingState();
+    return state == WorkingState.SUCCESSFUL || state == WorkingState.COMMITTED || state == WorkingState.FAILED;
```

Comment:
In `LocalJobManager.onTaskCompletion()`, it won't commit the job unless `jobState.getCompletedTasks() == jobState.getTaskCount()`. If a CANCELLED task doesn't count for completed tasks but does count for total task count, will it prevent the job from being committed?